pos_frequency_pipeline.py

```diff
@@ -10,16 +17,48 @@ class EmptyFileError(Exception):
 
 
 class POSFrequencyPipeline:
-    def __init__(self, corpus):
-        pass
+    def __init__(self, corpus_manager):
+        self.corpus_manager = corpus_manager
 
     def run(self):
-        pass
+        for article in self.corpus_manager.get_articles().values():
+            pos_frequencies = {}
+
+            article_path = article.get_file_path(ArtifactType.single_tagged)
+
+            with open(article_path, 'r') as text_file:
+                text = text_file.read()
+
+                if not text:
+                    raise EmptyFileError
+
+                tokens = text.split()
+
+                for token in tokens:
+                    match = re.search(r'<[A-Z]+', token)
+
+                    if not match:
+                        continue
+
+                    pos = match.group(0)[1:]
+                    pos_frequencies[pos] = pos_frequencies.get(pos, 0) + 1
+
+            with open(ASSETS_PATH / article.get_meta_file_path(), 'r', encoding='utf-8') as meta_file:
+                meta_data = json.load(meta_file)
+                meta_data.update({'pos_frequencies': pos_frequencies})
```

Comment:
should update happen inside context manager of file opening?

always try to keep file opening context manager clean: only logic related to getting text in a way you prefer and that is it